spy/doppler.py

```diff
@@ -475,3 +475,22 @@ def shift_expr_CallMethod(self, op: ast.CallMethod) -> ast.Expr:
         v_meth = self.shifted_expr[op.method]
         newargs_v = [self.shifted_expr[arg] for arg in op.args]
         return self.shift_opimpl(op, w_opimpl, [v_obj, v_meth] + newargs_v)
+
+    def shift_expr_AssignExpr(self, assignexpr: ast.AssignExpr) -> ast.Expr:
+        specialized = self.specialized_assignexprs[assignexpr]
+        new_value = self.shifted_expr[assignexpr.value]
+        if isinstance(specialized, ast.AssignLocal):
+            return ast.AssignExprLocal(
+                loc=assignexpr.loc,
+                target=assignexpr.target,
+                value=new_value,
+            )
+        elif isinstance(specialized, ast.AssignCell):
+            return ast.AssignExprCell(
+                loc=assignexpr.loc,
+                target=assignexpr.target,
+                target_fqn=specialized.target_fqn,
+                value=new_value,
+            )
```

Comment:
I think you can just use `specialize.replace(value=newvalue)` instead of having to go through all the cases manually, can't you?